renku/api/storage.py

```diff
@@ -53,6 +70,30 @@ class StorageApiMixin(RepositoryApiMixin):
 
     _CMD_STORAGE_PULL = ['git', 'lfs', 'pull', '-I']
 
+    @cached_property
+    def storage_installed(self):
+        """Verify that git-lfs is installed and on system PATH."""
+        return bool(which('git-lfs'))
+
+    @cached_property
+    def has_external_storage(self):
+        """Check if repository has external storage enabled.
+
+        :raises: ``errors.ExternalStorageNotInstalled``
+        :raises: ``errors.ExternalStorageDisabled``
+        """
+        repo_config = self.repo.config_reader(config_level='repository')
+        lfs_enabled = repo_config.has_section('filter "lfs"')
+
+        storage_enabled = lfs_enabled and self.storage_installed
+        if self.use_external_storage and not storage_enabled:
```

Comment:
With this the user has to specify `-S` flag with each command even when the project is initialised with `-S`. This looks a bit inconvenient to me.

Reply:
This is exactly the check we need, otherwise, user can by accident commit the large file and destroy the repo.